core/iwasm/common/wasm_memory.c

```diff
@@ -1389,20 +1389,20 @@ wasm_enlarge_memory_internal(WASMModuleInstanceCommon *module,
     if (full_size_mmaped) {
 #ifdef BH_PLATFORM_WINDOWS
         if (!os_mem_commit(memory->memory_data_end,
-                           (mem_offset_t)(total_size_new - total_size_old),
+                           (uint64)(total_size_new - total_size_old),
```

Comment:
these casts don't seem necessary to me.

Reply:
It's about a corner case:
```webassembly
(module
  (memory (;0;) 0)
  (export "main" (func 0))
  (func (;0;) (result)
    block (result) ;; label = @1
      i32.const 65536
      memory.grow
      i64.load align=2
      drop
    end
  )
)
```
When in wasm32, the mem_offset_t is uint32, and total_size_new - total_size_old will be 0x100000000 and overflow as 0. Resulting in incorrectly throwing memory OOB exception for hardware boundary check.

Reply:
i meant, because total_size_old and total_size_new are uint64, there are no point to cast it to uint64 again.
you can just remove the mem_offset_t casts.

Reply:
You are right! Good catch